standards/core/lightpush.md

```diff
@@ -0,0 +1,88 @@
+---
+slug: 19
+title: 19/WAKU2-LIGHTPUSH
+name: Waku v2 Light Push
+status: draft
+editor: Zoltan Nagy <zoltan@status.im> 
+contributors: 
+  - Hanno Cornelius <hanno@status.im>
+  - Daniel Kaiser <danielkaiser@status.im>
+  - Oskar Thorén <oskarth@titanproxy.com>
+---
+
+previous version: `/vac/waku/lightpush/2.0.0-beta1` [19/WAKU2-LIGHTPUSH](https://rfc.vac.dev/waku/standards/core/19/lightpush)
+
+---
+**Protocol identifier**: `/vac/waku/lightpush/2.0.0-beta2`
+
+## Motivation and Goals
+
+Light nodes with short connection windows and limited bandwidth wish to publish messages into the Waku network.
+Additionally, there is sometimes a need for confirmation that a message has been received "by the network"
+(here, at least one node).
+
+`19/WAKU2-LIGHTPUSH` is a request/response protocol for this.
+
+## Payloads
+
+```protobuf
+syntax = "proto3";
+
+message LightPushRequest {
+    enum RequestType { // Future extension point
+        RELAY = 0;
+    }
```

Comment:
In my mind we push a message to a node not to a protocol.

The only other protocol I can think of would be store.

:thinking: What would it mean to push to store instead of relay?

Reply:
@jm-clius  can explain I think, but mainly yes, push message to store but not relay or both and additionally the RFC refers to 

```forward the `LightPushRequest` via 19/LIGHTPUSH on a [WAKU2-DANDELION](https://github.com/waku-org/specs/blob/waku-RFC/standards/application/dandelion.md) stem```

Reply:
Wouldn't pushing to Relay with the `ephemeral` field on be the same as pushing to both Relay and Store?

Reply:
How I understood this, to request just a store of the message but not to relay. Suppose we have a light client where using a lightpush service to directly store a message. @jm-clius is this what you meant?

Reply:
> In my mind we push a message to a node not to a protocol.

Indeed, I think the point here is to allow for future differentiation of "message services" that you're requesting from the service node. In that sense it's not a list of protocols in the request type, but a list of possible services of which "relay publishing" may be one. Other possible services that we envision:
- store the message (without publishing to relay)
- dandelion hop (for anonymisation)
- add rln proof and publish (arguably a different service than just publishing)

However, @NagyZoltanPeter I think for this version of the protocol we can perhaps leave the enum undefined and just add a comment that tag/field number `10` is reserved for this use case? It would make more sense to add once we have more services.

Reply:
> > In my mind we push a message to a node not to a protocol.
> 
> Indeed, I think the point here is to allow for future differentiation of "message services" that you're requesting from the service node. In that sense it's not a list of protocols in the request type, but a list of possible services of which "relay publishing" may be one. Other possible services that we envision:
> 
> * store the message (without publishing to relay)
> * dandelion hop (for anonymisation)
> * add rln proof and publish (arguably a different service than just publishing)

I re-read and I feel like it those use cases could be new protocols not extension. What's the benefit of doing it this way? The obvious downside is the added complexity of having multiple "type" of services in one protocol.